src/network/address.rs

```diff
@@ -110,12 +110,169 @@ impl fmt::Debug for Address {
     }
 }
 
+/// Address received from BIP155 addrv2 message
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct AddrV2Message {
+    /// Time that this node was last seen as connected to the network
+    pub time: u32,
+    /// Service bits
+    pub services: ServiceFlags,
+    /// Network ID + Network Address
+    pub addr: AddrV2,
+    /// Network port, 0 if not applicable
+    pub port: u16
+}
+
+/// Supported networks for use in BIP155 addrv2 message
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub enum AddrV2 {
```

Comment:
nit: I think Rust convention is to not capitalize abbreviations in type/variant names. So `Ipv6`, `Cjdns` etc. I don't feel strongly about this, though.